client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js

```diff
@@ -1238,6 +1195,175 @@ var FolderToolbarView = Backbone.View.extend({
+                let collection_item = {};
+                let dataset = Galaxy.libraries.folderListView.folder_container.get("folder").get(checked_items[i]);
+                collection_item.id = checked_items[i];
+                collection_item.name = dataset.get("name");
+                collection_item.deleted = dataset.get("deleted");
+                collection_item.state = dataset.get("state");
+                collection_elements.push(collection_item);
+            }
+        } else if (elements_source === "folder") {
+            collection_elements = new Backbone.Collection(
+                Galaxy.libraries.folderListView.folder_container.get("folder").where({type: "file"})
+            ).toJSON();
+        }
+        let new_history_name = this.modal.$("input[name=history_name]").val();
+        if (new_history_name !== "") {
+            this.createNewHistory(new_history_name)
+                .done(new_history => {
+                    mod_toastr.success("History created");
+                    this.collectionImport(collection_elements, new_history.id, new_history.name);
+                })
+                .fail((xhr, status, error) => {
+                    mod_toastr.error("An error occurred.");
+                });
+        } else {
+            let selected_history_id = this.select_collection_history.value();
+            let selected_history_name = this.select_collection_history.text();
+            this.collectionImport(collection_elements, selected_history_id, selected_history_name);
+        }
+    },
+
+    collectionImport: function(collection_elements, history_id, history_name) {
+        let modal_title = `Creating Collection in ${history_name}`;
+        let creator_class;
+        let creationFn;
+        if (this.collectionType === "list"){
+            creator_class = LIST_CREATOR.ListCollectionCreator;
+            creationFn = (elements, name, hideSourceItems) => {
+                elements = elements.map(element => ({
+                    id: element.id,
+                    name: element.name,
+                    src: "ldda"
+                }));
+                return this.createHDCA(elements, this.collectionType, name, hideSourceItems, history_id);
+            };
+            LIST_CREATOR.collectionCreatorModal(
+            collection_elements,
+            {creationFn: creationFn, title: modal_title},
+            creator_class
+            );
+        } else if (this.collectionType === "paired"){
+            creator_class = PAIR_CREATOR.PairCollectionCreator;
+            creationFn = (elements, name, hideSourceItems) => {
+                elements = [
+                { name: "forward", src: "ldda", id: elements[0].id },
+                { name: "reverse", src: "ldda", id: elements[1].id }
+                ];
+                return this.createHDCA(elements, this.collectionType, name, hideSourceItems, history_id);
+            };
+            LIST_CREATOR.collectionCreatorModal(
```

Comment:
PAIR_CREATOR.pairCollectionCreatorModal right?

Reply:
The function you mention is just a wrapper over `LIST_CREATOR.collectionCreatorModal` that adds a title which I do not want to use. I can refactor it if you deem it a better approach.

Reply:
Should we just change the title everywhere then to the one you want - yours looks better? If yes, can we eliminate the other entries points all together?

Also, if the LIST_CREATOR has a good general purpose collection creator method we want to use to create pairs - can we move it into base-creator.js to make it more clear its function?

I know these are your problems - but I'm happy to clean this up in a follow up PR if you wish. I'd just like to better understand the collection creator entry points.

Reply:
>I'd just like to better understand the collection creator entry points.

you and me both 🤣

Reply:
@jmchilton The modal titles here are only different in the fact they include name of the history you are creating the collections in - which is only relevant for the datalib use case - when you are invoking the collection modals from history it is obvious which one you are creating it in.